new-log-viewer/src/components/Editor/MonacoInstance/language.ts

```diff
@@ -0,0 +1,36 @@
+import * as monaco from "monaco-editor/esm/vs/editor/editor.api.js";
+
+import {TOKEN_NAMES} from "./typings";
+
+
+const LOG_LANGUAGE_NAME = "logLanguage";
+
+
+/**
+ * Registers a custom log language in the Monaco editor.
+ */
+const setupCustomLogLanguage = () => {
+    monaco.languages.register({
+        id: LOG_LANGUAGE_NAME,
+    });
+    monaco.languages.setMonarchTokensProvider(LOG_LANGUAGE_NAME, {
+        tokenizer: {
+            root: [
+                /* eslint-disable @stylistic/js/array-element-newline */
+                ["INFO", TOKEN_NAMES.CUSTOM_INFO],
+                ["WARN", TOKEN_NAMES.CUSTOM_WARN],
+                ["ERROR", TOKEN_NAMES.CUSTOM_ERROR],
+                ["FATAL", TOKEN_NAMES.CUSTOM_FATAL],
+                [/(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3})Z/, TOKEN_NAMES.CUSTOM_DATE],
+                [/^[\t ]*at.*$/, TOKEN_NAMES.CUSTOM_EXCEPTION],
+                [/(\d+(?:\.\d+)?([eE])([+-])[0-9](\.[0-9])?|\d+(?:\.\d+)?)/, TOKEN_NAMES.CUSTOM_NUMBER],
```

Comment:
Line length violation. Any idea why this wasn't caught by @stylistic/js/max-len?

Reply:
By default, `ignoreRegExpLiterals`, which ignores lines that contain a RegExp literal, is true in the rule.

https://eslint.style/rules/js/max-len#:~:text=%22ignoreRegExpLiterals%22%3A%20true%20ignores%20lines%20that%20contain%20a%20RegExp%20literal